src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Connection.cs

```diff
@@ -138,11 +147,12 @@ public virtual void OnSocketClosed()
                         var connection2 = (Connection)state2;
                         connection2._filteredStream.Dispose();
                         connection2._adaptedPipeline.Dispose();
+                        Input.Reader.Complete();
                     }, connection);
                 }
             }, this);
 
-            Input.Dispose();
+            Input.Writer.Complete(new TaskCanceledException("The request was aborted"));
```

Comment:
Yes. To get to this code path, at least one of the following things happened:

1. The most likely scenario is that the app func completed.
2. The app called HttpRequestLifetimeFeature,Abort().
3. The app cancelled the token passed to Response.Body.WriteAsync().
4. A write failed.

The last one might be the most controversial reason for making reads fail, but we decided to kill the connection on write failures a long time ago. Even though subsequent writes won't fail if not passed the RequestAborted token, any subsequent reads will. It's been this way since at least 1.0.0.